06/Coordinatron.scala

```diff
@@ -1,129 +1,119 @@
-    absoluteValue(p1._1 - p2._1) + absoluteValue(p1._2 - p2._2)
-  }
-
-  def max(a: Int, b: Int): Int = {
-    if (a > b) a else b
-  }
-
-  def min(a: Int, b: Int): Int = {
-    if (a < b) a else b
+  def distance(p1: Point, p2: Point): Int = {
+    (p1.x - p2.x).abs + (p1.y - p2.y).abs
   }
 
   // finds the bounds as follows: x min, x max, y min, y max
-  def bounds(l: List[(Int, Int)]): (Int, Int, Int, Int) = {
+  def bounds(l: List[Point]): Bounds = {
     @annotation.tailrec
-    def go(l: List[(Int, Int)], xMin: Int, xMax: Int, yMin: Int, yMax: Int): (Int, Int, Int, Int) = {
-      if (l.isEmpty) (xMin, xMax, yMin, yMax)
-      else go(l.tail, min(l.head._1, xMin), max(l.head._1, xMax), min(l.head._2, yMin), max(l.head._2, yMax))
+    def go(l: List[Point], xMin: Int, xMax: Int, yMin: Int, yMax: Int): Bounds = {
+      if (l.isEmpty) Bounds(xMin, xMax, yMin, yMax)
+      else go(l.tail, l.head.x.min(xMin), l.head.x.max(xMax), l.head.y.min(yMin), l.head.y.max(yMax))
     }
-    go(l.tail, l.head._1, l.head._1, l.head._2, l.head._2)
+    go(l.tail, l.head.x, l.head.x, l.head.y, l.head.y)
   }
 
   // returns the list's closest point (or one of them), and true if equidistant to multiple points or false otherwise
-  def findClosestPoint(p: (Int, Int), l: List[(Int, Int)]): ((Int, Int), Boolean) = {
-    def go(p: (Int, Int), l: List[(Int, Int)], s: Int, c: (Int, Int), e: Boolean): ((Int, Int), Boolean) = {
-      if (l.isEmpty) (c, e)
+  def findClosestPoint(point: Point, l: List[Point]): MapPoint = {
+    def go(l: List[Point], shortestDistance: Int, closestPoint: Point, equidistant: Boolean): MapPoint = {
+      if (l.isEmpty) MapPoint(closestPoint, equidistant)
       else {
-        val currentDistance: Int = distance(p, l.head)
-        val shortest: Int = min(currentDistance, s)
-        val closer: (Int, Int) = if (shortest == s) c else l.head
-        /* if shortest doesn't equal s, it changed; equi must become false
-         if currentDistance equals s, we have seen this distance before; equi must become true
-         in all other cases equi must remain at the value of e */
-        val equidistant: Boolean = if (shortest != s) false else if (currentDistance == s) true else e
-        go(p, l.tail, shortest, closer, equidistant)
+        val currentDistance: Int = distance(point, l.head)
+        val shorter: Int = currentDistance.min(shortestDistance)
+        val closer: Point = if (shorter == shortestDistance) closestPoint else l.head
+        /* if shorter doesn't equal shortestDistance passed to this iteration of go, it changed;
+         equidistant must become false
+         if currentDistance equals shortestDistance, we have seen this distance before; equidistant must become true
+         in all other cases equidistant must remain at the value passed to this iteration of go */
+        val newEquidistant: Boolean = if (shorter != shortestDistance) false else if (currentDistance == shortestDistance) true else equidistant
```

Comment:
Constructions like `if(a) true else false` Could be replaced with just `a`.
If you have additional conditions, you might use `||` or `&&` for collapsing boolean expressions.
In this case the last `if` could be rewritten according to the following rule:
```scala
if(a) true else b ===> a || b
```
then the first if could be replaced:
```scala
if(a) false else b ===> !a && b
```
So you might get something like
```scala
val newEquidistant = shorter == shortestDistance && (currentDistance == shortestDistance || equidistant)
```